File: ts/src/errors/ValidationError.ts
/**
 * ValidationError represents a structural or integrity failure detected while a token is checked
 * against the Tokenforge specification.
 */
export class ValidationError extends Error {
    /**
     * Constructs a ValidationError carrying the given human-readable reason.
     */
    public constructor(reason: string) {
        super(reason);
        this.name = "ValidationError";

        // Restore the prototype chain so instanceof works after transpilation to ES5 targets.
        Object.setPrototypeOf(this, ValidationError.prototype);
    }
}

File: go/internal/fault/validation_error.go
package fault

// ValidationError represents a structural or integrity failure detected while a
// token is checked against the Keycutter specification.
type ValidationError struct {
	reason string
}

// NewValidationError creates a ValidationError carrying the given human-readable reason.
func NewValidationError(reason string) *ValidationError {
	return &ValidationError{reason: reason}
}

// Error returns the underlying failure reason and satisfies the error interface.
func (validationError *ValidationError) Error() string {
	return validationError.reason
}
